### DISCUSSION TITLE
Downgrade offered as upgrade

### DISCUSSION THREAD

**Discussion by tzugen** [renovatebot/renovate#17890] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

Latest, installed from Docker today

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

Gitlab.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Version 2.71 offered as upgrade to version 2.8:
https://gitlab.com/ultrasonic/ultrasonic/-/merge_requests/831

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction, but I have linked to a public repo where it occurs

**Comment by rarkins** [comment]
`71828` > `8`, so it sorts higher and is considered an upgrade.

Looking at [gradle versioning](https://docs.gradle.org/current/userguide/single_versions.html), I don't see why it should be treated like `2.7.1` as you seem to expect.

Would you also consider 2.10 to be less than 2.8?

**Reply by rarkins** [reply]
It's also sorted highest here: https://mvnrepository.com/artifact/com.squareup.picasso/picasso

![image](https://user-images.githubusercontent.com/6311784/191345539-a29e3e61-a7b2-47b1-afc7-ea9767a05e89.png)
